### ISSUE TITLE
Browser version broken in 0.2.28

### ISSUE THREAD

**Issue by oliver-moran** [oliver-moran/jimp#274]
@strandedcity, dammit the browser version has broken again in this release. Any ideas what is up with it? I know too it is being used by many projects so this is a bit of a catastrophe.

**Comment by strandedcity** [comment]
eep! Sorry for the delay on this end I'll look into it ASAP. Had a baby ~recently, so time has been short! But I'll find some to take a look :)

It seems like we've proven the build process in its current state is on the fragile side when dependencies change. I'll try to think of something more stable, or at least a way to quickly test breakages before they occur (ideally on the PR level).

Sorry for the delay -- I'm not gone! Just busy.

**Comment by strandedcity** [comment]
Sorry to say the browser build is now completely unfamiliar to me... @aurium rewrote it over a series of steps:

https://github.com/oliver-moran/jimp/commits/master/tools/browser-build.js

In particular the error I now see when npm tries to prepublish makes perfect sense if the file is interpreted as es5, not es6:

```
> jimp@0.2.28 prepublish /Users/phil/PROJECTS/jimp
> node tools/browser-build.js prepublish

/Users/phil/PROJECTS/jimp/tools/browser-build.js:4
var {normalize, join} = require('path');
```

@aurium do you expect ```$ npm run prepublish``` to work still? It doesn't for me. If you expect this to work now differently, I'm certain @oliver-moran  would appreciate re-alignment of the new build process with package.json.

Out of curiosity, what's the current size of the minified browser file? The new build looks more purposeful and robust than my original one, but I did a lot of work to keep the file as small as it could possibly be....

**Comment by aurium** [comment]
@strandedcity `$ npm run prepublish` should work with no problem.

Your problem with `Unexpected token {` may be your node.js version. The node version and tools demands es6 features. Don't wore about the browser. it is transpiled do es5.

**Comment by strandedcity** [comment]
which version of node is required?

```
$ node --version
v5.1.0
```
If a particular version of node is now required for jimp, we should include it in package.json:

```
{ "engines" : { "node" : ">=0.10.3 <0.12" } }
```

https://docs.npmjs.com/files/package.json

However I expect that doing so may break some people's server-side projects. So perhaps the better way to go is to check the node version in the script itself so we can throw a more precise error eg "You need nodejs version 6.x.x or greater to run the browser build":

https://stackoverflow.com/questions/6656324/check-for-current-node-version

**Comment by strandedcity** [comment]
@oliver-moran upgrading node to the current stable version (8.0.0) worked for me; the build runs. I haven't yet had a chance to test the output, but it runs. Were you stuck where I got stuck?

```
$ node --version
v8.0.0
```

**Comment by hipstersmoothie** [comment]
This will be fixed in #473
